src/tasks/cosplay/array/list/MyArrayList.java

```diff
@@ -0,0 +1,121 @@
+package tasks.cosplay.array.list;
+
+import java.util.Arrays;
+
+public class MyArrayList<T> implements MyList<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size = 0;
+
+    public MyArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void arrayGrower() {
+        int tempLength = array.length + (array.length >> 1);
+        T[] temp = (T[]) new Object[tempLength];
+        System.arraycopy(array, 0, temp, 0, array.length);
+        array = temp;
+    }
+
+    @Override
+    public void add(T value) {
+        if (size == array.length) {
+            arrayGrower();
+        }
+        add(value, size);
+    }
+
+    @Override
+    public void add(T value, int index) {
+        addIndexValidation(index);
+        T[] temp = (T[]) new Object[size - index];
+        System.arraycopy(array, index, temp, 0, temp.length);
+        array[index] = value;
+        System.arraycopy(temp, 0, array, index + 1, temp.length);
+        size++;
+    }
+
+    @Override
+    public void addAll(MyList<T> myList) {
+        if (myList != null) {
+            while (size + myList.size() >= array.length) {
+                arrayGrower();
+            }
+            for (int i = 0; i < myList.size(); i++) {
+                add(myList.get(i));
+            }
+        }
+    }
+
+    @Override
+    public T get(int index) {
+        indexValidation(index);
+        return array[index];
+    }
+
+    @Override
+    public void set(T value, int index) {
+        indexValidation(index);
+        array[index] = value;
+    }
+
+    @Override
+    public T remove(int index) {
+        indexValidation(index);
+        T value = array[index];
+        System.arraycopy(array, index + 1, array, index, size() - index);
+        size--;
+        return value;
+    }
+
+    @Override
+    public T remove(T t) {
+        int index = 0;
+        for (int i = 0; i < size; i++) {
+            if (t.equals(array[i])) {
+                index = i;
+            }
+        }
+        return remove(index); // Yes, array[size-1] and array[size] are the same,
```

Comment:
better try to resolve this problem
we can use Arrays.copyOf() to remove this value